src/Intersection.jl

```diff
@@ -472,6 +477,14 @@ function ρ(d::AbstractVector{N},
     return ρ_helper(d, swap(cap), algorithm; kwargs...)
 end
 
+# disambiguation
+function ρ(d::AbstractVector{N},
+           cap::Intersection{N, S1, S2}) where {N<:Real,
+           S1<:Union{HalfSpace{N}, Hyperplane{N}, Line{N}},
+           S2<:Union{HalfSpace{N}, Hyperplane{N}, Line{N}}}
+    return ρ(d, HPolyhedron([cap.X, cap.Y]))
```

Comment:
This does not make sense. An `HPolyhedron` cannot be constructed from a list of sets.

Reply:
😄  I was assuming the `cap.X` and ` cap.Y` are `HalfSpace`. But the other cases need to be handled as well...

Reply:
We can do `HPolyhedron([constraints_list(cap.X); constraints_list(cap.Y)])`.